plumbing/format/gitignore/README.md

```diff
@@ -0,0 +1,71 @@
+# `gitignore` parser and matcher for Go
```

Comment:
Instead of a README.md we document the packages as package documentation as:
https://github.com/src-d/go-git/blob/master/plumbing/format/index/doc.go

And this appears at godoc like: https://godoc.org/gopkg.in/src-d/go-git.v4/plumbing/format/index